pkg/deploy/spindeploy/transformer/patches_test.go

```diff
@@ -0,0 +1,164 @@
+			},
+		},
+		{
+			"one strategic merge as yaml",
+			`
+apiVersion: spinnaker.io/v1alpha2
+kind: SpinnakerService
+metadata:
+  name: spinnaker
+spec:
+  kustomize:
+    gate:
+      deployment:
+        patchesStrategicMerge:
+          - |
+            spec:
+             selector:
+               matchLabels:
+                 newlabel: blah 
+`,
+			func(t *testing.T, dep *appsv1.Deployment) {
+				assert.Equal(t, "blah", dep.Spec.Selector.MatchLabels["newlabel"])
+			},
+		},
+		{
+			"one json merge as yaml",
+			`
+apiVersion: spinnaker.io/v1alpha2
+kind: SpinnakerService
+metadata:
+  name: spinnaker
+spec:
+  kustomize:
+    gate:
+      deployment:
+        patches:
+          - |
+            spec:
+              template:
+                spec:
+                  containers:
+                  - name: other
+`,
+			func(t *testing.T, dep *appsv1.Deployment) {
+				if assert.Equal(t, 1, len(dep.Spec.Template.Spec.Containers)) {
+					assert.Equal(t, "other", dep.Spec.Template.Spec.Containers[0].Name)
+				}
+			},
+		},
+		{
+			"one strategic merge as yaml - add a container",
+			`
+apiVersion: spinnaker.io/v1alpha2
+kind: SpinnakerService
+metadata:
+  name: spinnaker
+spec:
+  kustomize:
+    gate:
+      deployment:
+        patchesStrategicMerge:
+          - |
+            spec:
+              template:
+                spec:
+                  containers:
+                  - name: other
+`,
+			func(t *testing.T, dep *appsv1.Deployment) {
+				assert.Equal(t, 2, len(dep.Spec.Template.Spec.Containers))
+			},
+		},
+		{
+			"json6902 patch as yaml",
+			`
+apiVersion: spinnaker.io/v1alpha2
+kind: SpinnakerService
+metadata:
+  name: spinnaker
+spec:
+  kustomize:
+    gate:
+      deployment:
+        patchesJson6902: |
+          - op: replace
+            path: /spec/template/spec/containers/0/image
+            value: nginx 
+`,
+			func(t *testing.T, dep *appsv1.Deployment) {
+				if assert.Equal(t, 1, len(dep.Spec.Template.Spec.Containers)) {
+					assert.Equal(t, "nginx", dep.Spec.Template.Spec.Containers[0].Image)
+				}
+			},
+		},
+		{
+			"json6902 patch as yaml, remove volume mount",
+			`
+apiVersion: spinnaker.io/v1alpha2
+kind: SpinnakerService
+metadata:
+  name: spinnaker
+spec:
+  kustomize:
+    gate:
+      deployment:
+        patchesJson6902: |
+          - op: remove
+            path: /spec/template/spec/containers/0/volumeMounts/1
+`,
+			func(t *testing.T, dep *appsv1.Deployment) {
+				if assert.Equal(t, 1, len(dep.Spec.Template.Spec.Containers)) {
+					assert.Equal(t, 1, len(dep.Spec.Template.Spec.Containers[0].VolumeMounts))
+				}
+			},
+		},
+	}
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			_, spinsvc := th.setupTransformerFromSpinText(&exposeLbTransformerGenerator{}, c.spinsvc, t)
+			gen := &generated.SpinnakerGeneratedConfig{}
+			test.AddDeploymentToGenConfig(gen, "gate", "testdata/input_deployment.yml", t)
+			p := patchTransformer{svc: spinsvc}
```

Comment:
I think there's no need for this line, since the above method call `th.setupTransformerFromSpinText` in line 154 already produces the instance of the transformer to be used, just provide the `patchTransformerGenerator`, which by the way, implicitly tests the generator itself that seemed to have a bug ;)